syntax = "proto3";

package ocr;

service OCRService {
    rpc ExtractData(OCRRequest) returns (OCRResponse);
}

message OCRRequest {
    bytes image_data = 1;
    string version = 2;  // "2018" or "2025"
    string side = 3;     // "recto" or "verso"
    bool thinking_mode = 4; // Enable thinking mode in prompts
}

message OCRResponse {
    bool success = 1;
    string raw_text = 2;
    string extracted_data = 3;  // JSON string
    string error = 4;
}